### PR TITLE
Support soft limit in NewPooledTransactions

### LINKED ISSUE #1484: Support soft limit in NewPooledTransactions
### Describe the feature

The soft limit for NewPooledTransactionhashes is 4096
https://github.com/ethereum/devp2p/blob/master/caps/eth.md#newpooledtransactionhashes-0x08

this is mostly an issue for the first newpooledHashes message:

https://github.com/paradigmxyz/reth/blob/3c41807f0f7e92df1ebd5f5ebc2ffa4d542185dc/crates/net/network/src/transactions.rs#L341-L342

which fetches _all_ hashes from the pool

## TODO
* enforce soft limit for NewPooledTransactions

blocked by #1483 #1482

### Additional context

https://github.com/paradigmxyz/reth/pull/1483

### PR DESCRIPTION
**Pull request by Tirthnp**
Fixes #1484

### CODE REVIEW

**Comment by Tirthnp** [comment]
Not sure if this is right, please let me know :)

File: crates/net/network/src/transactions.rs [hunk 1]
```diff
@@ -547,12 +547,19 @@ enum PooledTransactionsHashesBuilder {
 
 impl PooledTransactionsHashesBuilder {
     fn push(&mut self, tx: &PropagateTransaction) {
+        let soft_limit = 4096;
```
Comment (mattsse) [inline crates/net/network/src/transactions.rs:550]: pleas make this another constant, see top of file.

File: crates/net/network/src/transactions.rs [hunk 2]
```diff
@@ -547,12 +547,19 @@ enum PooledTransactionsHashesBuilder {
...
         match self {
-            PooledTransactionsHashesBuilder::Eth66(msg) => msg.0.push(tx.hash()),
+            PooledTransactionsHashesBuilder::Eth66(msg) => {
+                if msg.0.len() <= soft_limit {
```
Comment (mattsse) [inline crates/net/network/src/transactions.rs:553]: this check should happen where `.push` is called,
see `todo!` in file.

**Review by mattsse — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]
lgtm,

please enable edits by maintainer for next contribution

**Pull request merged**